enterprise-suite/Makefile

```diff
@@ -1,27 +1,91 @@
-test: lint
+# Using ?= cuz enterprise-suite-latest overrides this
+CHART ?= enterprise-suite
 
-lint: lint-helm lint-json lint-promql
+RELEASE_NAME = es
+NAMESPACE = lightbend
 
-lint-helm:
-	helm lint .
+define banner
+	$(info === $@)
+endef
 
-lint-json:
-	find . -name \*.json | xargs -tn 1 jq . >/dev/null
+# Note: We may want to require explicit lists.  As implemented, we may end
+# up with random cruft in the packages...
+COMPONENTS := $(wildcard */.)
+SUBCOMPONENTS := $(wildcard */*/.)
 
-lint-promql:
-	scripts/validate-promql.sh
+HELM_CHARTS_DIR := ..
+SCRIPTS_DIR := $(HELM_CHARTS_DIR)/scripts
 
-minikube-test:
-	scripts/run-minikube-tests.sh
+#####
+# Note:  These are tweaked by the hacky enterprise-suite-latest Makefile...
+VERSION ?= $(shell $(SCRIPTS_DIR)/export-chart-version.sh $(CHART))
+RELEASE = $(CHART)-$(VERSION)
+CHART_DIR = .
+ALL_YAML ?= all.yaml
+#####
 
-delete-local:
-	-helm delete --purge es
 
-install-local:
-	helm install ../docs/$(RELEASE).tgz --name=es --namespace=lightbend --debug --wait
+all: lint build  ## lint then build chart
+build: init $(HELM_CHARTS_DIR)/docs/$(RELEASE_NAME)/$(ALL_YAML)  ## Build chart package (and all.yaml)
 
-install-local-latest:
-	helm install ../docs/$(RELEASE_LATEST).tgz --name=es --namespace=lightbend --debug --wait
+# This target required by top-level helm-charts Makefile.  Tests release on Travis
+test:
+	$(CHART_DIR)/scripts/run-minikube-tests.sh
 
-.PHONY: test lint lint-helm lint-json lint-promql minikube-test \
-	delete-local install-local install-local-latest
+# We no longer need a target to build the index.yaml file since that's up to the
+# helm-charts Makefile.  Could include this though...
+# build: $(HELM_CHARTS_DIR)/docs/index.yaml
+#
+# $(HELM_CHARTS_DIR)/docs/index.yaml: $(HELM_CHARTS_DIR)/docs/$(RELEASE).tgz
+# 	$(call banner)
+# 	( cd .. ; $(MAKE) -C $(HELM_CHARTS_DIR) docs/index.yaml ; )
+
+init:
+	@$(SCRIPTS_DIR)/lib.sh
+	@helm init -c > /dev/null
+
+$(HELM_CHARTS_DIR)/docs/$(RELEASE_NAME)/$(ALL_YAML): $(HELM_CHARTS_DIR)/docs/$(RELEASE).tgz
```

Comment:
Leaving the building of `all.yaml` in for now since the front-end guys apparently rely on it at this time.  We can easily yank it out later if need be.